### PR TITLE
fix(data-check): explain mid-epoch supply gaps instead of reporting value creation

### PR DESCRIPTION
**Pull request by scarmuega**
When a pool pays its registration deposit during an epoch, the live UTxO pot changes before the boundary-valued pool count does. `dolos data check` now accounts for that gap instead of reporting valid state as created or destroyed supply. Any unexplained residual still fails.

The checker excludes retired pools from live deposit accounting, recognizes rolling activity even in a slot-zero block, and reports a skipped assertion only when nonzero lovelace is in flight. Untouched boundaries and zero-value gaps remain exact checks. The CLI summary and module documentation explain the distinction.

Validation on PR head `9259a8df`:

- Formatter and full workspace build pass.
- Default workspace suite: 1,511 passed, 48 ignored.
- All-features workspace suite with the repository's three service exclusions: 1,026 passed, 53 ignored.
- Regressions cover retired pools, boundary corruption, first-block registration, explained deposits, zero-count/zero-value deposits, and an extra one-lovelace discrepancy.
- Replacement mainnet store `mainnet-v17-stelae-20260910`, tip `197512829`: all five consistency checks passed on `306e7439`. After the final zero-value-only adjustment, totals passed again on the final source in 9.79 seconds; the other four checks are unchanged. This store has no deposit gap, so no skip is expected. Positive-gap reporting is covered synthetically.

Remaining QA gate: full-workspace Clippy completes with seven pre-existing warnings in unchanged Cardano and snapshot test files. `-D warnings` fails on those warnings; the warning-free requirement has not been waived. The plan records the required scope decision for their mechanical cleanup.

Plan: `plans/dolos-data-check-mid-epoch-supply.md`. Merge remains with the founder after QA completion.

### CODE REVIEW

File: src/bin/dolos/data/check/totals.rs [hunk 1]
```diff
@@ -521,22 +542,86 @@ pub fn check_pots(claimed: &Pots, found: &Recomputed, max_supply: Option<u64>) -
...
+            } else {
+                not_assertable.push(NotAssertable::new(
+                    CHECK,
+                    format!(
+                        "exact supply is not assertable at a mid-epoch tip: \
+                         {in_flight_pool_count} pool deposit(s), totaling {in_flight_lovelace} \
+                         lovelace, are absent from the boundary-valued pool pot; after accounting \
+                         for them the genesis supply matches",
+                    ),
+                ));
```
[the remark this answers is not in the document]
Comment (scarmuega) [reply src/bin/dolos/data/check/totals.rs:631]: Fixed in 306e7439. A zero in-flight pool count no longer emits NotAssertable. The regression checks exact equality and an additional one-lovelace discrepancy, which still fails. All 76 checker tests and both required workspace suites pass.

File: src/bin/dolos/data/check/totals.rs [hunk 2]
```diff
@@ -1002,13 +1088,44 @@ pub fn run(
...
+            Some(max_supply),
+        ) => {
+            let (_, epoch_slot) = summary.slot_epoch(slot);
+            if epoch_slot == 0 {
+                Some(SupplyCheck::Exact { max_supply })
+            } else {
+                Some(SupplyCheck::MidEpoch {
+                    max_supply,
+                    live_pool_count: found.registered_pools,
+                })
```
[the remark this answers is not in the document]
Comment (scarmuega) [reply src/bin/dolos/data/check/totals.rs]: Fixed in 306e7439. Exact mode now requires epoch slot zero and no live rolling statistics. A regression runs first-block pool-deposit statistics through mode selection, live_pots, and check_pots; it verifies the deposit is explained and an extra lovelace still fails. Another test keeps an untouched boundary strict. Both required workspace suites pass.

File: src/bin/dolos/data/check/totals.rs [hunk 3]
```diff
@@ -521,22 +558,85 @@ pub fn check_pots(claimed: &Pots, found: &Recomputed, max_supply: Option<u64>) -
...
+                    CHECK,
+                    format!(
+                        "the mid-epoch pots plus {in_flight_pool_count} in-flight pool deposit(s) \
+                         ({in_flight_lovelace} lovelace) account for {accounted_supply} lovelace, \
+                         but genesis fixes the supply at {max_supply} (off by {}); the pool \
+                         deposits do not explain the discrepancy",
+                        accounted_supply.abs_diff(max_supply),
+                    ),
+                ));
+            } else if in_flight_pool_count > 0 {
```
[the remark this answers is not in the document]
Comment (scarmuega) [reply src/bin/dolos/data/check/totals.rs]: Fixed in 9259a8df. NotAssertable now requires positive in-flight lovelace. The new zero_cost_pool_registration_does_not_skip_exact_supply regression covers a positive pool-count delta with a zero deposit, including an extra one-lovelace discrepancy that still fails. The regression passes in the default workspace suite; the full QA rerun is continuing.

**Comment by scarmuega** [comment]
QA published the preserved retired-pool fix `8f7884d2`, the slot-zero/zero-count fix `306e7439`, and the follow-up zero-value-deposit fix `9259a8df`. All three actionable bot threads are addressed and resolved; the duplicate review-body findings share those fixes.

Final validation: formatter and workspace build pass; default tests pass (1,511), and the prescribed all-features suite passes (1,026). The replacement mainnet store passed all five checks on `306e7439`, followed by a passing totals check on the final source after the zero-value-only change. Archive continuity took 860.40 seconds; final totals took 9.79 seconds. The store reports tip `197512829` consistently across WAL, archive, and state.

The generic request to reach 80% docstring coverage is rejected as an ungrounded style threshold: the repository does not mandate it. Contract documentation and fixture rationale are preserved; comment normalization removed one repeated inline supply-policy comment.

QA handoff remains pending the plan's warning-free lint gate. The complete Clippy command returns success with seven warnings already present in the unchanged base: one `filter_next`, two `cloned_ref_to_slice_refs`, and four `doc_lazy_continuation` warnings in a snapshot-test doc paragraph. Strict `-D warnings` fails. Their mechanical cleanup is an explicit scope decision in `plans/dolos-data-check-mid-epoch-supply.md`; no acceptance criterion has been waived.
